File: python/day-01.py
import os


def part1():
    input = get_input()
    return [(b > a) for a, b in zip(input, input[1:])].count(True)


def part2():
    input = get_input()
    return [(b > a) for a, b in zip(input, input[3:])].count(True)


def get_input():
    input = open(os.path.join(os.path.dirname(__file__),
                              '../inputs/day-01.txt'), 'r').read()
    return list(map(int, input.splitlines()))


if __name__ == "__main__":
    print('Part 1: ' + str(part1()))
    print('Part 1: ' + str(part2()))

File: deno/day-01.ts
console.log('Part 1:', part1())
console.log('Part 2:', part2())

function part1() {
  // 1374
  const input = getInput().split('\n').map(parseFloat)
  return zip(input, input.slice(1)).filter(([a, b]) => b > a).length
  // imperative
  // return input.reduce((count, reading, index) => {
  //   if (index === 0) return count
  //   return reading > input[index - 1] ? ++count : count
  // }, 0)
}

function part2() {
  const input = getInput().split('\n').map(parseFloat)
  return zip(input, input.slice(3)).filter(([a, b]) => b > a).length
}

function zip<A, B>(a: A[], b: B[]): Array<[A, B]> {
  return a.map((itemA, i) => [itemA, b[i]])
}

function getInput(): string {
  return Deno.readTextFileSync('../inputs/day-01.txt')
}
